### PR TITLE
[DataGrid] Fix falsy filter values not showing in filter button tooltip

### PR DESCRIPTION
**Pull request by ithrforu**
Displaying before fix:
https://codesandbox.io/s/muddy-star-18il32?file=/demo.tsx

After:
https://codesandbox.io/s/determined-ellis-hidp5r?file=/demo.tsx

### CODE REVIEW

File: packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx [hunk 1]
```diff
@@ -1105,8 +1105,88 @@ describe('<DataGrid /> - Filter', () => {
     });
   });
 
+  describe('filter button tooltip\'s displaying values', () => {
```
Comment (MBilalShafi) [inline packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx:1108]: Maybe cover both usecases in a single test?
```diff
diff --git a/packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx b/packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx
index f7ee5337c..e96f2f374 100644
--- a/packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx
+++ b/packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx
@@ -1105,35 +1105,39 @@ describe('<DataGrid /> - Filter', () => {
     });
   });
 
-  describe('filter button tooltip\'s displaying values', () => {
-    it('should `false` value display', () => {
-      render(
-        <div style={{ width: '100%', height: '400px' }}>
-          <DataGrid
-            filterModel={{
-              items: [{ id: 0, field: 'isAdmin', operator: 'is', value: false }],
-            }}
-            rows={[
-              {
-                id: 0,
-                isAdmin: false,
-              },
-            ]}
-            columns={[
-              {
-                field: 'isAdmin',
-                type: 'singleSelect',
-                valueOptions: [
-                  {
-                    value: false,
-                    label: false,
-                  },
-                ],
-              },
-            ]}
-            components={{ Toolbar: GridToolbarFilterButton }}
-          />
-        </div>,
+  describe('filter button tooltip', () => {
+    it('should display `falsy` value', () => {
+      const { setProps } = render(
+        <DataGrid
+          filterModel={{
+            items: [{ id: 0, field: 'isAdmin', operator: 'is', value: false }],
+          }}
+          autoHeight
+          rows={[
+            {
+              id: 0,
+              isAdmin: false,
+              level: 0,
+            },
+          ]}
+          columns={[
+            {
+              field: 'isAdmin',
+              type: 'singleSelect',
+              valueOptions: [
+                {
+                  value: false,
+                  label: false,
+                },
+              ],
+            },
+            {
+              field: 'level',
+              type: 'number',
+            },
+          ]}
+          components={{ Toolbar: GridToolbarFilterButton }}
+        />,
       );
 
       const filterButton = document.querySelector('button[aria-label="Show filters"]')!;
@@ -1146,41 +1150,8 @@ describe('<DataGrid /> - Filter', () => {
 
       expect(tooltip).toBeVisible();
       expect(tooltip.textContent).to.contain('false');
-    });
 
-    it('should `falsy` value display', () => {
-      render(
-        <div style={{ width: '100%', height: '400px' }}>
-          <DataGrid
-            filterModel={{
-              items: [{ id: 0, field: 'level', operator: '=', value: 0 }],
-            }}
-            rows={[
-              {
-                id: 0,
-                level: 0,
-              },
-            ]}
-            columns={[
-              {
-                field: 'level',
-                type: 'number',
-              },
-            ]}
-            components={{ Toolbar: GridToolbarFilterButton }}
-          />
-        </div>,
-      );
-
-      const filterButton = document.querySelector('button[aria-label="Show filters"]')!;
-      expect(screen.queryByRole('tooltip')).to.equal(null);
-
-      fireEvent.mouseOver(filterButton);
-      clock.tick(1000); // tooltip display delay
-
-      const tooltip = screen.getByRole('tooltip');
-
-      expect(tooltip).toBeVisible();
+      setProps({ filterModel: { items: [{ id: 1, field: 'level', operator: '=', value: 0 }] } });
       expect(tooltip.textContent).to.contain('0');
     });
   });
```

File: packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx [hunk 2]
```diff
@@ -93,7 +93,9 @@ const GridToolbarFilterButton = React.forwardRef<HTMLButtonElement, GridToolbarF
                 <li key={index}>
                   {`${lookup[item.field!].headerName || item.field}
                   ${getOperatorLabel(item)}
-                  ${item.value ? getFilterItemValue(item) : ''}`}
+                  ${
+                    item.value !== undefined && item.value !== null ? getFilterItemValue(item) : ''
```
Comment (MBilalShafi) [inline packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx:97]: Wouldn't this cover both `undefined` and `null` cases?
```suggestion
                    item.value != null ? getFilterItemValue(item) : ''
```

**Review by MBilalShafi — COMMENTED** [review]
Thank you for identifying and fixing this. I've added a couple of recommendations.

File: packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx (hunk 2 above)
Comment (ithrforu) [reply packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx:97]: I think implicit conversions are worse than explicit ones.

Comment (MBilalShafi) [reply packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx:97]: You are right, in some cases like the comparison of numbers with strings, implicit comparisons might not work great (Example: 12 == "12" ✅) 

IMO though, for falsy values, using this shortcut could be a useful way to keep the code more readable especially if the intention is only to check for `undefined` or `null` (as no other falsy value like `0` or `false` would coerce to be `null`) 🤷‍♂️

Comment (ithrforu) [reply packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx:97]: I'll use `item.value != null` with comment about it.

**Comment by ithrforu** [comment]
@MBilalShafi, thank you for good recommendations. All done :)

**Review by MBilalShafi — APPROVED** [review]
Thank you for the contribution 🙏

**Pull request merged**